### PR TITLE
CortexXDR adding an option to retrieve file without specify the os type

### PR DESCRIPTION
**Pull request by orhovy**
Due to Github issues opening this PR for Orel Haim.
For any Q please reach out to her.

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/34265

## Description
Adding an option to retrieve file without specify the os type.

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation 

@EliorKedar @michalgold FYI.

### CODE REVIEW

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 1]
```diff
@@ -973,17 +973,31 @@ def get_endpoint_device_control_violations(self, endpoint_ids: list, type_of_vio
 
         return reply.get('reply')
 
-    def retrieve_file(self, endpoint_id_list: list, windows: list, linux: list, macos: list) -> Dict[str, Any]:
+    def generate_files_dict_with_specific_os(self, windows: list, linux: list, macos: list):
```
Comment (bakatzir) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:976]:
```suggestion
    def generate_files_dict_with_specific_os(self, windows: list, linux: list, macos: list) -> Dict[str, list] :
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 2]
```diff
@@ -973,17 +973,31 @@ def get_endpoint_device_control_violations(self, endpoint_ids: list, type_of_vio
...
-            files['linux'] = macos
+            files['macos'] = macos
 
-        if not windows and not linux and not macos:
-            raise ValueError('You should enter at least one path.')
+        return files
+
+    def retrieve_file(self, endpoint_id_list: list, windows: list, linux: list, macos: list, file_path_list: list)\
+            -> Dict[str, Any]:
+        # there are 2 options, either the paths are given with separation to a specific os ire without
```
Comment (bakatzir) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:992]:
```suggestion
        # there are 2 options, either the paths are given with separation to a specific os or without
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 3]
```diff
@@ -1004,6 +1018,33 @@ def retrieve_file(self, endpoint_id_list: list, windows: list, linux: list, maco
...
         return reply.get('reply')
 
+    def generate_files_dict(self, endpoint_id_list: list, file_path_list: list) -> Dict[str, Any]:
+        files: dict = {"windows": [], "linux": [], "macos": []}
+
+        if len(endpoint_id_list) != len(file_path_list):
+            raise ValueError("The endpoint_ids list must be in the same length as the generic_file_path")
+
+        for endpoint_id, file_path in zip(endpoint_id_list, file_path_list):
+            endpoint = self.get_endpoints(endpoint_id_list=[endpoint_id])
```
Comment (bakatzir) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:1028]:
```suggestion
            endpoints = self.get_endpoints(endpoint_id_list=[endpoint_id])
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 4]
```diff
@@ -1004,6 +1018,33 @@ def retrieve_file(self, endpoint_id_list: list, windows: list, linux: list, maco
...
+
+            if len(endpoint) == 0:
+                raise ValueError(f'Error: Endpoint {endpoint_id} was not found')
+
+            endpoint = endpoint[0]
```
Comment (bakatzir) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:1033]:
```suggestion
            endpoint = endpoints[0] if isinstance(endpoints, list) else endpoints
```

**Review by bakatzir — CHANGES_REQUESTED** [review]
Please also test the case where `Error: Endpoint {endpoint_id} was not found`

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 5]
```diff
@@ -2070,6 +2070,15 @@ script:
       name: mac_file_paths
       required: false
       secret: false
+    - default: false
+      description: A comma-separated list of file paths in any platform. Can be used
+        instead of the mac/windows/linux file paths. The order of the files path list
+        must be parellel to the endpoints list order, therefore, the first file path
+        in the list is related to the first endpoint and so on.
```
Comment (bakatzir) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:2077]: an e.g would be great here.

Comment (bakatzir) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:2077]:
```suggestion
        in the list is related to the first endpoint and so on, e.g: "C:\Users\demisto\Desktop\CortexXSOAR.txt".
```

**Review by bakatzir — APPROVED** [review]
Looks good. @daryakoval please have a look as well as you've added this functionality recently.

**Review by daryakoval — APPROVED** [review]

**Pull request merged**